src/quarot/rtn_utils.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+# Based on the implementation in https://github.com/spcl/QuaRot/
+
+import torch
+import tqdm
+
+from slicegpt import utils
+
+from .quant_utils import WeightQuantizer, find_qlayers
+
+torch.backends.cuda.matmul.allow_tf32 = False
+torch.backends.cudnn.allow_tf32 = False
+
+
+@torch.no_grad()
+def rtn_fwrd(model, dev, args):
+    '''
+    Round-to-nearest quantization of the model weights.
+    '''
+    assert args.w_groupsize == -1, "Groupsize not supported in RTN!"
+    layers = model.model.layers
+    torch.cuda.empty_cache()
+
+    quantizers = {}
+
+    for i in tqdm.tqdm(range(len(layers)), desc="(RtN Quant.) Layers"):
+        layer = layers[i].to(dev)
+
+        subset = find_qlayers(layer, layers=[torch.nn.Linear])
```

Comment:
@sashkboos could you explain why we're trying to find linear blocks that haven't been wrapped with the ActQuantWrapper here?

Reply:
@nailimixaM 

You can do it over the ActQuantWrapper as well. The whole point is to pass the weight matrices to the quantizer and round them.